cekit-helpers/cekit-cache-server/src/main/java/org/kie/cekit/cacher/utils/CacherUtils.java

```diff
@@ -266,6 +266,38 @@ public boolean fileExistsByNameExcludeTmp(String fileName) {
         }
     }
 
+    /**
+     * Verifies if the given file exists on the persisted files, excludes tmp dir from the search
+     *
+     * @param fileName to be searched
+     * @return List of found {@link PlainArtifact}
+     */
+    public List<PlainArtifact> getFilesByName(String fileName) {
+        Path path = FileSystems.getDefault().getPath(cacherProperties.getCacherArtifactsDir());
+
+        List<PlainArtifact> artifacts = new ArrayList<>();
+
+        try {
+            artifacts = Files.walk(path)
+                    .filter(Files::isRegularFile)
+                    .filter(file -> file.getFileName().toString().contains(fileName))
+                    .map(p -> {
+                        try {
+                            return new PlainArtifact(p.getFileName().toString(),
+                                    p.getParent().getFileName().toString(),
+                                    Files.getAttribute(p.toAbsolutePath(), "creationTime", LinkOption.NOFOLLOW_LINKS).toString(),
+                                    null, null, null);
+                        } catch (IOException e) {
+                            e.printStackTrace();
+                            return null;
+                        }
+                    }).collect(Collectors.toList());
```

Comment:
Do we want to have null values in the artifacts (as they can be returned in catch block)? I think it would be good to filter them out, so we can avoid NPE issues.